src/mkFlake.nix

```diff
@@ -0,0 +1,135 @@
+iogx-inputs:
+
+let
+
+  utils = import ./boot/utils.nix iogx-inputs;
+  modularise = import ./boot/modularise.nix iogx-inputs;
+  options = import ./boot/options.nix iogx-inputs;
+
+
+  mkNixpkgs = system: args:
+    import iogx-inputs.nixpkgs {
+      inherit system;
+      config = iogx-inputs.haskell-nix.config // args.config;
+      overlays =
+        [
+          # WARNING: The order of these is crucial
+          iogx-inputs.iohk-nix.overlays.crypto
+          iogx-inputs.iohk-nix.overlays.cardano-lib
+          iogx-inputs.haskell-nix.overlay
+          iogx-inputs.iohk-nix.overlays.haskell-nix-crypto
+          iogx-inputs.iohk-nix.overlays.haskell-nix-extra
+        ]
+        ++ args.overlays;
+    };
+
+
+  mkIogxLib = desystemized-user-inputs: pkgs:
+    let
+      iogx = { inherit utils modularise options; };
+      repoRoot = modularise {
+        root = ../.;
+        module = "repoRoot";
+        args = {
+          user-inputs = desystemized-user-inputs;
+          iogx-inputs = utils.deSystemize pkgs.stdenv.system iogx-inputs;
+          lib = builtins // pkgs.lib // { inherit iogx; };
+          system = pkgs.stdenv.system;
+          inherit pkgs;
+        };
+      };
+    in
+    {
+      mkShell = repoRoot.src.core.mkShell;
+      mkHaskellProject = repoRoot.src.core.mkHaskellProject;
+      mkHydraRequiredJob = repoRoot.src.core.mkHydraRequiredJob;
+
+      inherit (iogx) utils modularise options;
+    };
+
+
+  mkFlake =
+    { inputs
+    , repoRoot
+    , systems ? [ "x86_64-linux" "x86_64-darwin" ]
+    , outputs ? [ ]
+    , flake ? { }
+    , nixpkgsArgs ? { config = { }; overlays = [ ]; }
+    , debug ? false
+    , ...
+    }:
```

Comment:
Thanks for the suggestion @coderabbitai.
The argument to `mkFlake` can and will be turned into a nix module so that we can avail of the nix module system to do type checking

Reply:
@coderabbiat that's pretty good, but can you turn the `systems` field into an enum? 
Permitted values are [ "x86_64-linux" "x86_64-darwin"  "aarch64-linux" "aarch64-darwin"]

Reply:
@coderabbitai that's pretty good, but can you turn the `systems` field into an enum? 
Permitted values are [ "x86_64-linux" "x86_64-darwin"  "aarch64-linux" "aarch64-darwin"]